dissect/target/plugins/os/unix/linux/proc.py

```diff
@@ -0,0 +1,715 @@
+        return self.parent.name
+
+    @property
+    def state(self) -> str:
+        """Returns the state of the process (S'leeping, R'unning, I'dle, etc)."""
+        return ProcessStateEnum[self._stat_file.get("state")].value
+
+    @property
+    def starttime(self) -> datetime:
+        """Returns the start time of the process."""
+        # Starttime is saved in clockticks per second from the boot time.
+        # we asume a standard of 100 clockticks per second. the actual value can be obtained from `getconf CLK_TCK`
+        starttime = self._stat_file.get("starttime") / 100
+
+        return from_unix(self._boottime + starttime)
+
+    @property
+    def runtime(self) -> timedelta:
+        """Returns the runtime of a process until the moment of acquisition."""
+        runtime = self.now - self.starttime
+        return runtime
+
+    @property
+    def now(self) -> datetime:
+        """Returns the `now()` timestamp of the system at the moment of acquisition."""
+        now = self.uptime + from_unix(self._boottime)
+        return now
+
+    def environ(self) -> Iterator[Environ]:
+        """Yields the content of the environ file associated with the process as variable name and value pairs.
+
+        Yields:
+            An iterator containing a tuple with the variable name and value thereof.
+        """
+        yield from self._parse_environ()
+
+    @property
+    def uptime(self) -> datetime:
+        """Returns the uptime of the system from the moment it was acquired."""
+        # uptime is saved in seconds from boottime
+        uptime = timedelta(seconds=float(self.target.fs.get("/proc/uptime").open().readline().split()[0]))
+
+        return uptime
+
+    @property
+    def _process_name(self) -> str:
+        """Internal function that returns the name of the process."""
+        return self._stat_file.get("comm", None).decode()
+
+    @property
+    def cmdline(self) -> str:
+        """Return the command line of a process."""
+        cmdline = self.get("cmdline").open().readline().decode()
+        # Cmdlines are null-terminated and use null-bytes to separate the different parts. Translate this back.
+        cmdline = cmdline.rstrip("\x00").replace("\x00", " ")
+        return cmdline
+
+    def stat(self) -> fsutil.stat_result:
+        """Return a stat entry of the process."""
+        return self.entry.stat()
+
+
+class ProcPlugin(Plugin):
+    __namespace__ = "proc"
+
+    def __init__(self, target: Target) -> None:
+        super().__init__(target)
+        # make this a lookup instead of a pre-processed map
+        self.inode_map = None
+        self.sockets = Sockets(self.target)
+
+    def _iter_proc_pids(self) -> Iterator[Tuple[str, FilesystemEntry]]:
+        """Yields /proc/<pid> filesystems entries for every process id (pid) found in procfs
+
+        Yields:
+            An iterator containing the process id (pid) and corrensponding filesystem entry.
+        """
```

Comment:
```suggestion
        """Yields ``/proc/<pid>`` filesystems entries for every process id (pid) found in procfs."""
```
You also use `<pid>` here but `[pid]` in some of the docstrings above.